src/components/PostDetails.tsx

```diff
@@ -1,117 +1,136 @@
+import { warningTimer } from '../utils/warningTimer';
+
+type Props = {
+  post: Post | null;
+  comments: Comment[];
+  setComments: Dispatch<SetStateAction<Comment[]>>;
+  isLoaderComments: boolean,
+};
+
+export const PostDetails: React.FC<Props> = ({
+  post,
+  comments,
+  setComments,
+  isLoaderComments,
+}) => {
+  const [isWriteComment, setIsWriteComment] = useState(false);
+  const [isHasError, setIsHasError] = useState(false);
+
+  const removeComment = async (commentRemoveId: number) => {
+    try {
+      await deleteComment(commentRemoveId);
+      setComments(prevComments => prevComments
+        .filter(({ id }) => id !== commentRemoveId));
+    } catch {
+      setIsHasError(true);
+      warningTimer(setIsHasError, false, 3000);
+    }
+  };
+
+  useEffect(() => {
+    setIsWriteComment(false);
+  }, [post]);
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            {`${post?.id}: ${post?.title}`}
           </h2>
 
           <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
+            {post?.body}
           </p>
         </div>
 
         <div className="block">
-          <Loader />
+          {isLoaderComments && <Loader />}
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
-
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
-
-          <p className="title is-4">Comments:</p>
-
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
+          {isHasError && (
+            <div className="notification is-danger" data-cy="CommentsError">
+              Something went wrong
             </div>
+          )}
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
+          {!isLoaderComments && !isHasError && (!comments.length ? (
+            <p className="title is-4" data-cy="NoCommentsMessage">
+              No comments yet
+            </p>
+          ) : (
+            <>
+              <p className="title is-4">Comments:</p>
+              {comments.map(comment => {
```

Comment:
you can object destructuring right here
```suggestion
              {comments.map(({id, email, name, body}) => {
```